### ISSUE TITLE
Parallel Execution

### ISSUE THREAD

**Issue by ekristen** [ekristen/aws-nuke#191]
## Overview

Cleanup could potentially be done faster by doing parallel execution. Discovery of resources is already done in parallel to an extent.

## Considerations

- Deleting resources controlled by CFN can cause issues with CFN deleting.
- CFN can take a long time to delete, perhaps sending to background the remove tasks is possible.

## Implementation

This would actually be done in [libnuke](https://github.com/ekristen/libnuke). There were recently some changes made for GCP to add an additional optional wait handler to the resource type for when the backend doesn't give an immediate response, it could potentially be leveraged to background CFN removals.

## Upstream References

- https://github.com/rebuy-de/aws-nuke/issues/1230
